### PR TITLE
Link to model detail page in errors

### PR DESCRIPTION
**Pull request by rosbo**
Before:

```
You don't have permission to access resource at URL: https://www.kaggle.com/api/v1/models/keras/g_mini/keras/g_mini_2b_en/download
Please make sure you are authenticated if you are trying to access a private resource or a resource requiring consent.
```

After:

```
You don't have permission to access resource at URL: https://www.kaggle.com/models/keras/g_mini/frameworks/keras/variations/g_mini_2b_en
Please make sure you are authenticated if you are trying to access a private resource or a resource requiring consent.
```

http://b/323204229

### CODE REVIEW

**Comment by rosbo** [comment]
cc/ @mayankmalik-colab You may want to do something similar to ensure people on Colab get a nice error message.

File: src/kagglehub/clients.py [hunk 1]
```diff
@@ -120,7 +121,7 @@ def post(self, path: str, data: dict) -> dict:
             self._check_for_version_update(response)
             return response_dict
 
-    def download_file(self, path: str, out_file: str) -> None:
+    def download_file(self, path: str, out_file: str, resource_handle: Optional[ResourceHandle] = None) -> None:
```
Comment (mohami2000) [inline src/kagglehub/clients.py:124]: not sure if this is on purpose, but you didnt change how download_file is getting called anywhere in the code. is the change for the future if needed?
download is called in `http_resolver.py` and `test_kaggle_api_client`

Comment (rosbo) [reply src/kagglehub/clients.py:124]: Good catch, I actually forgot to commit my latest change 🤦

Comment (rosbo) [reply src/kagglehub/clients.py:124]: Can you also review the refactor PR please: https://github.com/Kaggle/kagglehub/pull/79. Thank you

Comment (mohami2000) [reply src/kagglehub/clients.py:124]: i commented there as well

**Review by mohami2000 — APPROVED** [review]

**Pull request merged**